public/app/plugins/datasource/grafana-azure-monitor-datasource/components/LogsQueryEditor/QueryField.tsx

```diff
@@ -31,26 +31,28 @@ const QueryField: React.FC<AzureQueryEditorFieldProps> = ({ query, datasource, o
   }
 
   useEffect(() => {
-    const promises = [
-      datasource.azureLogAnalyticsDatasource.getKustoSchema(query.azureLogAnalytics.workspace),
-      getPromise(),
-    ] as const;
+    if (query.azureLogAnalytics.resource) {
```

Comment:
the git diff here is a bit hard to see but the only change is that I wrapped this in an if statement

Reply:
what do you think about inverting the if statement?

```typescript
if (!query.azureLogAnalytics.resource) {
  return
}
```

😌